### PR TITLE
Changed unittest to pytest. Added name to contributing.

### PR DESCRIPTION
**Pull request by datamel**
test_config.py updated to pytest.

### CODE REVIEW

File: metomi/rose/tests/test_config.py [hunk 1]
```diff
@@ -20,102 +20,100 @@
...
-        conf.set(["dinner", "dessert"], "custard")
-        self.assertEqual(list(iter(conf)), ["food", "dinner"])
-        end_node = conf.get(["", "food"])
-        self.assertEqual(list(iter(end_node)), [])
-
-
-class TestConfigDump(unittest.TestCase):
+
+
+def test_init():
```
Comment (wxtim) [inline metomi/rose/tests/test_config.py]: Might be nice to have a more informative name - `test_rose_config_init`, perhaps?

File: metomi/rose/tests/test_config.py [hunk 2]
```diff
@@ -20,102 +20,100 @@
...
+    assert(node.value == {})
+    node = conf.get(["rubbish"])
+    assert(node is None)
+    assert(conf.unset(["rubbish"]) is None)
+
+
+def test_set():
+    """Test setting/unsetting value/ignored flag in a Config object."""
+    conf = metomi.rose.config.ConfigNode()
+    assert(conf is not None)
```
Comment (wxtim) [inline metomi/rose/tests/test_config.py]: Happy about the removal of a double negative here.

File: metomi/rose/tests/test_config.py [hunk 3]
```diff
@@ -20,102 +20,100 @@
...
+    assert((node.value, node.state) == ("value", "!!"))
+    assert(conf.unset(["rubbish"]) is not None)
+    conf.set(["rubbish"], {})
+    node = conf.get(["rubbish"])
+    assert((node.value, node.state) == ({}, ""))
+    conf.set(["rubbish", "item"], "value")
+    node = conf.get(["rubbish", "item"])
+    assert((node.value, node.state) == ("value", ""))
+    conf.get(["rubbish"]).state = "!"
+    assert(conf.get(["rubbish", "item"], True) is None)
```
Comment (wxtim) [inline metomi/rose/tests/test_config.py]: Checked these and they all seem logically equiv. Happy days.

File: metomi/rose/tests/test_config.py [hunk 4]
```diff
@@ -20,102 +20,100 @@
...
+def test_iter():
+    """Test the iterator"""
+    conf = metomi.rose.config.ConfigNode()
+    conf.set(["", "food"], "glorious")
+    conf.set(["dinner", "starter"], "soup")
+    conf.set(["dinner", "dessert"], "custard")
+    assert(list(iter(conf)) == ["food", "dinner"])
+    end_node = conf.get(["", "food"])
+    assert(list(iter(end_node)) == [])
+
```
Comment (wxtim) [inline metomi/rose/tests/test_config.py]: Checked test method. Happy with this one.

File: metomi/rose/tests/test_config.py [hunk 5]
```diff
@@ -20,102 +20,100 @@
...
+    assert((node.value, node.state) == ({}, ""))
+    conf.set(["rubbish", "item"], "value")
+    node = conf.get(["rubbish", "item"])
+    assert((node.value, node.state) == ("value", ""))
+    conf.get(["rubbish"]).state = "!"
+    assert(conf.get(["rubbish", "item"], True) is None)
+
+
+def test_iter():
+    """Test the iterator"""
```
Comment (sadielbartholomew) [inline metomi/rose/tests/test_config.py]:
```suggestion
    """Test the iterator."""
```

**Comment by datamel** [comment]
This closes issue #2308

**Pull request merged**